pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -63,56 +62,29 @@ public static boolean isEqual(Object[] left, Object[] right) {
     return Arrays.deepEquals(left, right);
   }
 
-  public static boolean isEqual(Collection left, Collection right) {
-    if (left != null && right != null) {
-      return left.toString().equals(right.toString());
+  public static boolean isEqual(Object left, Object right) {
+    if (left != null) {
+      return left.equals(right);
     } else {
-      return left == right;
+      return right == null;
     }
   }
 
+  @SuppressWarnings("unchecked")
   public static boolean isEqualIgnoringOrder(List left, List right) {
     if (left != null && right != null) {
-      ArrayList sortedLeft = new ArrayList(left);
-      ArrayList sortedRight = new ArrayList(right);
+      List sortedLeft = new ArrayList(left);
+      List sortedRight = new ArrayList(right);
       Collections.sort(sortedLeft);
       Collections.sort(sortedRight);
-      return sortedLeft.toString().equals(sortedRight.toString());
-    } else {
-      return left == right;
-    }
-  }
-
-  public static boolean isEqual(Map left, Map right) {
```

Comment:
For all classes that extend Object, use it's own equals() is good enough. I will add a unit test to test some basic data structure.